### PR TITLE
Adding stubs for FoundryClient for backcompat

### PR DESCRIPTION
**Pull request by pashakondratyev**
Adds stubs for `FoundryClient` and generating a back-compat interface. 
Here we're using a generated client to instantiate with the correct types from the metadata.

### CODE REVIEW

File: packages/generator/src/v1.1/generateClientSdkVersionOneDotOne.test.ts [hunk 1]
```diff
@@ -49,6 +52,9 @@ describe("generator", () => {
       );
     }
 
-    expect(diagnostics).toHaveLength(0);
+    // TODO: Certain errors are expected since we can't resolve the static code, but we should fix them.
+    const errors = diagnostics.filter((q) => q.code !== 2792);
```
Comment (pashakondratyev) [inline packages/generator/src/v1.1/generateClientSdkVersionOneDotOne.test.ts:56]: These are related to the imports in the generated code

File: .vscode/settings.json [hunk 2]
```diff
@@ -9,5 +9,8 @@
   },
   "[json]": {
     "editor.defaultFormatter": "dprint.dprint"
+  },
+  "editor.codeActionsOnSave": {
```
Comment (pashakondratyev) [inline .vscode/settings.json:13]: Are there any strong opinions on adding this? I can add this to my user-settings instead

File: packages/legacy-client/src/metadata/OntologyDefinition.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { LocalDate } from "../ontology-runtime/baseTypes";
+
+export interface OntologyDefinition<K extends string> {
```
[the remark this answers is not in the document]
Comment (ericanderson) [reply packages/legacy-client/src/metadata/OntologyDefinition.ts:19]: Yeah as we move forward we should consider how much we can consolidate.

File: .vscode/settings.json (hunk 2 above)
Comment (ericanderson) [reply .vscode/settings.json:13]: I generally lean against forced on save actions. @mfedderly thoughts?

**Review by ericanderson — APPROVED** [review]

**Comment by ericanderson** [comment]
Going to merge and we can handle follow ups separately if needed. Im happy for us to err on the side of moving quick.

File: .vscode/settings.json (hunk 2 above)
Comment (mfedderly) [reply .vscode/settings.json:13]: I'm generally pro having robots do stuff for me. We have to pass eslint to get code merged anyhow so we should at least do this as a precommit hook. We can also turn this setting off super easily later if we don't like it.

**Pull request merged**